### PR TITLE
(feat): Add pre-appointment outreach module

### PR DESCRIPTION
**Pull request by kipchubett**
- Adds weekly pre-appointment outreach module ~ patient-list

### CODE REVIEW

File: src/app/clinic-dashboard/general/pre-appointment-outreach/pre-appointment-outreach.component.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  public loadingpreAppointmentOutreachList = false;
+
+  public weeks: any[] = [];
+  public selectedWeek: string;
+  public selectedFormattedWeek: string;
+
+  // constants
+  private ALL = 'All';
+  private FOLLOW_UP_SUCCESSFUL = 'Follow-up Successful';
+  private FAILED_FOLLOW_UP_ATTEMPT = 'Failed Follow-up Attempt';
```
Comment (drizzentic) [inline src/app/clinic-dashboard/general/pre-appointment-outreach/pre-appointment-outreach.component.ts:34]: Would it be better if we used enum for the constants e.g
enum Status {
  ALL = 'All',
  FOLLOW_UP_SUCCESSFUL = 'Follow-up Successful',
  FAILED_FOLLOW_UP_ATTEMPT = 'Failed Follow-up Attempt'
}

Comment (kipchubett) [reply src/app/clinic-dashboard/general/pre-appointment-outreach/pre-appointment-outreach.component.ts:34]: At this point we aren't sure if these the only filter options, that said let's refactor this while doing the enhancements!

**Review by Alfred-Mutai — COMMENTED** [review]
LGTM

**Review by drizzentic — APPROVED** [review]

**Comment by kimaina** [comment]
@drizzentic / @corneliouzbett when will this be deployed? Dumish is planning to use this feature next week on monday

**Pull request merged**